core/entities/variant.go

```diff
@@ -81,7 +76,13 @@ func (v *variant) GetValue(ctx context.Context) string {
 
 func (v *variant) InternalGetValue(ctx context.Context) (returnValue string, isDefault bool) {
 	returnValue, isDefault = v.defaultValue, true
-	mergedContext := context.NewMergedContext(v.globalContext, ctx)
+	var mergedContext context.Context = nil
+
+	if v.parser != nil {
+		mergedContext = context.NewMergedContext(v.parser.GetGlobalContext(), ctx)
+	} else {
+		mergedContext = context.NewMergedContext(nil, ctx)
```

Comment:
won't context.NewMergedContext(nil, ctx) be the same as just using ctx ?
so
var mergedContext context.Context = ctx
if v.parser != nil { mergedContext = context.NewMergedC... }
?